crates/services/txpool_v2/src/service.rs

```diff
@@ -323,6 +324,29 @@ where
             let mut block_height = self.current_height.write();
             *block_height = new_height;
         }
+
+        // Remove expired transactions
+        let mut removed_txs = vec![];
+        {
+            let mut height_expiration_txs = self.pruner.height_expiration_txs.write();
+            let range_to_remove = height_expiration_txs
+                .range(..=new_height)
+                .map(|(k, _)| *k)
+                .collect::<Vec<_>>();
+            for height in range_to_remove {
+                let expired_txs = height_expiration_txs.remove(&height);
+                if let Some(expired_txs) = expired_txs {
+                    let mut tx_pool = self.pool.write();
+                    removed_txs
+                        .extend(tx_pool.remove_transaction_and_dependents(expired_txs));
+                }
+            }
+        }
+        for tx in removed_txs {
```

Comment:
would be nice to know if `removed_txs` are unique

Reply:
Yeah removed_txs should be unique in any cases because we don't want to push removal notification twice for the same id (it should'nt happens but who knows). I don't know if changing `removed_txs` to `Hashset` is ok because we iterate over it and might be performance consuming